packages/syft/src/syft/core/tensor/autodp/single_entity_phi.py

```diff
@@ -376,55 +377,57 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(entity={self.entity.name}, child={self.child})"
         )
 
+    # Check for shape1 = (1,s), and shape2 = (,s) --> as an example
     def __eq__(self, other: SupportedChainType) -> SingleEntityPhiTensor:
-
-        if is_acceptable_simple_type(other) or self.child.shape == other.child.shape:  # type: ignore
-            # if the tensor being compared is also private
-            if isinstance(other, SingleEntityPhiTensor):
-                if self.entity != other.entity:
-                    # this should return a GammaTensor
-                    return NotImplemented
-                data = self.child == other.child
+        if is_acceptable_simple_type(other):
+            if isinstance(other, np.ndarray):
+                # If other is a Numpy Array, we need to check if shapes can be broadcast
+                if is_broadcastable(other.shape, self.child.shape):
+                    data = self.child == other
+                else:
+                    raise Exception(
+                        f"Tensor shapes do not match for __eq__: {self.child.shape} != {other.child.shape}"
+                    )
             else:
-                # this can still fail, if shape1 = (1,s), and shape2 = (,s) --> as an example
                 data = self.child == other
-            min_vals = self.min_vals * 0.0
-            max_vals = self.max_vals * 0.0 + 1.0
-            entity = self.entity
-            return SingleEntityPhiTensor(
-                child=data,
-                entity=entity,
-                min_vals=min_vals,
-                max_vals=max_vals,
-                scalar_manager=self.scalar_manager,
-            )
-        else:
-            raise Exception(
-                f"Tensor shapes do not match for __eq__: {len(self.child)} != {len(other.child)}"  # type: ignore
-            )
-
-    def logical_and(self, other: SupportedChainType) -> SingleEntityPhiTensor:
-        if is_acceptable_simple_type(other) or self.child.shape == other.child.shape:
```

Comment:
I'm confused about what's going on here.

Reply:
if `other` is an acceptable_simple_type (int, float, bool, np.ndarray) then there are two scenarios:

- if it's a float/int/bool, then the comparison is doable, and no additional checks need to be made
- if it's a numpy array, then we need to check that self.child.shape and other.child.shape are broadcastable. If it's not, we raise an exception and tell the user that __eq__ can't be performed because the shapes are not broadcastable. If they are, we save the result of `self.child == other.child`, and return a SingleEntityPhiTensor whose child equals this result.

Sorry if my changes and thought process isn't clear! I can add more comments inline to explain.

Reply:
Ah makes sense - actualy what confuse dme was the way Github printed it. The code is clear.